rpcs3/Emu/Cell/SPUCommonRecompiler.cpp

```diff
@@ -33,6 +34,21 @@ const extern spu_decoder<spu_iflag> g_spu_iflag;
 
 constexpr u32 s_reg_max = spu_recompiler_base::s_reg_max;
 
+template <typename T>
+bool operator<(const std::span<T>& this_, const std::span<T>& that)
+{
+	return std::lexicographical_compare(this_.begin(), this_.end(), that.begin(), that.end());
```

Comment:
You can use std::memcmp or something like that for SPU data comparison, its just bits.. there is no reason for it to preserve this bwhaviour it only needs to detect unequality properly for ordered containers.

Reply:
I hope I understood you correctly. I changed it to use `std::memcmp` and do a comparison that does not take the span length into account.

Reply:
What I mean is, remove this operator overload and perform this code in SPU specifically.

Reply:
I removed it. If it is still not correct, just push to this PR. It's faster than trying to explain it.